tests/functional/testSubscription.js

```diff
@@ -87,6 +88,77 @@ describe('Subscription manager', function() {
       }
     };
 
+    it('Mock Desktop Authentication listener success', async () => {
```

Comment:
The test name `Mock Descktop Authentication listener success` has two problems:
1) it is describing an implementation detail of the test not the functional feature we're trying to test (this all started with wanting to make sure authenticationCompleted was successfully triggered).
2) it's not following the style conventions of a test name.

Here's some more context so you can pick another name (please let me know if this doesn't make sense):

The mocha framework is trying to lead you in the grammar with `describe() it()`. For example

`describe('My Choir') it('sings the national anthem when the queen enters the room.')`

See the next tests' name  is `it(Prompts SubscriptionNeeded if a user\'s subscription is inactive on connection)` inside the describe('Subscription Manager') describe ('Expired Subscription')`.

So, firstly, I'm not sure this test is specific to expired subscriptions so you may want to move this out block out of inside "Expired Subscription"... I think the double nested describe is unnecessary so would be fine with removing it. We don't have so many tests in here that we need to get overly fussy about grouping them.

I would suggest a test name like `describe('Subscription Manger') it(`Returns user to Main Screen after user successfully completes Web Subscription flow`). There's lots of other possibilities hopefully you get the idea.

Long test names are a feature not a bug.

Reply:
Thank you, I agree the current test name reflects the underlying implementation more than what the test is doing. I like your suggestion to rename it to `Returns user to Main Screen after user successfully completes Web Subscription flow`